test/jdk/com/sun/jdi/ConnectedVMs.java

```diff
@@ -76,9 +96,9 @@ void vms(int expected) {
     protected void runTests() throws Exception {
         System.out.println("Testing " + passName);
         vms(0);
-        startToMain("InstTarg");
-        ThreadReference thread = waitForVMStart();
-        StepEvent stepEvent = stepIntoLine(thread);
+        BreakpointEvent bp = startToMain("InstTarg");
+        waitForVMStart();
+        StepEvent stepEvent = stepIntoLine(bp.thread());
```

Comment:
These changes were needed for virtual thread support. `startToMain("InstTarg")` causes the debuggee to run until it it is suspended at a breakpoint in `InstTarg.main()`.  `waitForVMStart()` will return right away since the VM has already started, and will return the main thread of the debuggee, but this is the thread running `TestScaffold.main()`, which started up `InstTarg.main()` in a virtual thread. If we single step in the main thread in this case, the single step is not in `InstTarg.main()` as it should be, but is instead in main thread, which is blocked in the `join()` call waiting for the virtual thread to complete. The single step resumes all threads, but can't complete until the virtual thread exits. So before the test ever gets to do the `Process.destroy()`, `InstTarg.main()` has already exited

Fortunately it was easy to find the proper thread to single step in, since the virtual thread is the `BreakpointEvent` thread.